### PR TITLE
feat: Allow hide resources

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Related to #3450 

Depends on: https://github.com/coder/terraform-provider-coder/pull/47

### CODE REVIEW

File: provisionersdk/proto/provisioner.proto [hunk 1]
```diff
@@ -109,6 +109,7 @@ message Resource {
         bool is_null = 4;
     }
     repeated Metadata metadata = 4;
+	bool hide = 5;
```
Comment (deansheather) [inline provisionersdk/proto/provisioner.proto:112]:
```suggestion
    repeated Metadata metadata = 4;
    bool hide = 5;
```

**Review by deansheather — COMMENTED** [review]
You should also integrate the frontend changes in this PR too

**Comment by BrunoQuaresma** [comment]
> You should also integrate the frontend changes in this PR too

Yes, I will! I just want to check if the BE side of things is good.

**Review by code-asher — COMMENTED** [review]
Frontend looks great!  I ran into this error when trying to deploy:

```
migrate up: up: migration failed: column "hide" of relation "workspace_resources" contains null values in line 0: ALTER TABLE workspace_resources
    ADD COLUMN hide boolean NOT NULL;                                                                                                            
 (details: pq: column "hide" of relation "workspace_resources" contains null values)          
```

File: coderd/database/migrations/000047_workspace_resource_hide.up.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE workspace_resources
+    ADD COLUMN hide boolean NOT NULL;
```
Comment (coadler) [inline coderd/database/migrations/000047_workspace_resource_hide.up.sql:2]: Whenever you add a new column it needs to have a defualt value, like `DEFAULT 'f'`.

Comment (BrunoQuaresma) [reply coderd/database/migrations/000047_workspace_resource_hide.up.sql:2]: 100%

Comment (BrunoQuaresma) [reply coderd/database/migrations/000047_workspace_resource_hide.up.sql]: Fixed

**Comment by code-asher** [comment]
Nevermind ran `update_template_versions.sh` and all is well.

File: site/src/components/Resources/Resources.tsx [hunk 3]
```diff
@@ -51,129 +53,152 @@ export const Resources: FC<React.PropsWithChildren<ResourcesProps>> = ({
...
+
+      {hasHideResources && (
+        <div className={styles.buttonWrapper}>
+          <Button
+            className={styles.showMoreButton}
+            variant="outlined"
+            size="small"
+            onClick={() => setShouldDisplayHideResources((v) => !v)}
+          >
+            Show hidden resources{" "}
```
Comment (code-asher) [inline site/src/components/Resources/Resources.tsx:196]: Thoughts on changing this text when the hidden resources are showing?  My reasoning is that the button will say "show resources" but clicking on it actually hides them.  Maybe we can toggle between `show hidden resources` and `hide resources`?

**Review by code-asher — APPROVED** [review]

File: site/src/components/Resources/Resources.tsx (hunk 3 above)
Comment (BrunoQuaresma) [reply site/src/components/Resources/Resources.tsx:196]: Makes sense!

**Pull request merged**